lib/errors.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Append the old error message to the new one and keep its stack trace.
+ * Example:
+ *     throw wrap(e, HighLevelError, "This error is more specific");
+ */
+function wrap(
+    old_error,
+    new_error_type,
+    ...args
+) {
+    const new_error = new new_error_type(...args);
+    let append_msg;
+    if (old_error instanceof Error) {
+        append_msg = old_error.message;
+        new_error.stack = old_error.stack;
+    }
+    else {
+        append_msg = old_error.toString();
+    }
+    new_error.message += ":\n" + append_msg;
+    return new_error;
+}
+
+/**
+ * Sets the default message for the given error arguments.
+ */
+function default_message(args, msg) {
+    if (!(0 in args)) {
+        args[0] = msg;
+   }
+}
+
+/**
+ * Base Error for when the bride can not handle the event.
+ */
+class EventNotHandledError extends Error {
+    constructor(...args) {
+        default_message(args, "The event could not be handled by the bridge");
```

Comment:
You can also do `args[0] = args[0] || "This is my default message"`

The `||` operator is magical :)

Reply:
Oh…we're not doing Python here :smile: Python would make you suffer for that^^°

I'd argue to keep the method, as it makes the intent clearer. (I would still clean up `default_message` then.)